rfcs/2021-01-22-spatial-system-rfc.md

```diff
@@ -0,0 +1,204 @@
+
+- **Soft Grid:** Here everything still follows the 8-pt grid but we remove the baseline and horizontal grid line to a strict `8x8` grid. This creates a uniform spacing between elements which is always divisible by 4 or 8 which also helps developers on the implementation side.
+### **Multi-platform design**
+Also, we might find few screens where it is difficult to organise the spacing. For example, iPhone 6 (`375x667`). For such scenarios we don't need to break the pixels, instead we need to keep the margins & paddings consistent. It's ok to have oddly sized layout which keeps the grid consistent. Although, no user will likely see the actual measurements here.
+
+![Hard vs Soft Grid](./images/spatial/hard-soft-grid.png)
+
+>**📝Note:** We'll use the **Soft Grid** across all our products.
+
+### **Drawbacks**
+There are no major drawbacks as such, although we need to make sure that all the spacing between the elements should not be hardcoded. They should use the spatial tokens whenever required.
+
+# Alternatives
+There are many other spatial systems we might have considered, such as **5pt grid**, **6pt grid**. But we are not using them because of already mentioned reason,
+- With odd numbers like 5, it is hard to align things to center as the outcome will result in fraction values that will split the pixels. This can also result in blurry pixel.
+- Similarly the scaling will also not work properly while scaling on mobile/tablets, which will again result into pixel blurriness.
+
+# Adoption strategy
+Spatial scale needs to be implemented as a core part of the DS in terms of the tokens which will automatically derive the adoption.
+- Implement the [spatial scale](#spatial-scale) (as tokens).
+- Implement the 3 types of [column-grid](#column-grid) structure using the same spatial scale.
+- Make sure all the **components** are using the same spatial scale tokens while building them.
+- Every **product icon** is being made using the same spatial system canvas as defined inside Figma.
+- The `line-height` for **typography** should follow the same spatial system (of 8-pt grid) so that it aligns well with each other.
+
+# How do we educate people?
+Both **designers** & **developers** need to be educated.
+- Designers should always use the spatial (multiple of 4pt) scale while working on any layout.
+- Similarly, developers should always use the spatial tokens while defining the space between the adjacent elements inside the UI which should not be a problem as it is coming from a designer.
+
+# Open Questions
+- We can name the tokens in a better way, just like how we have did for colors.
+- We already have static tokens (i.e. which works for both desktops & mobiles) in place, should be need dynamic tokens that changes with the device?
+
+# References
```

Comment:
can we add [this video](https://www.youtube.com/watch?v=aPwwv-6pRWY) in references?